Joedmin-hw3/Joedmin-hw3/Program.cs

```diff
@@ -0,0 +1,153 @@
+
+                            while (true)
+                            {
+                                string[] credentials = GetCredencials();
+
+                                if (usersAndPasswords.ContainsKey(credentials[0]))
+                                {
+                                    if (!(string.Equals(credentials[1], usersAndPasswords[credentials[0]])))
+                                        Console.WriteLine("Invalid password. Please try again");
+                                    else
+                                    {
+                                        Console.WriteLine("Hello!");
+                                        break;
+                                    }
+                                }
+                                else
+                                    Console.WriteLine("Username is not registered");
+                            }
+                        }
+                    }
+                    catch (FileNotFoundException)
+                    {
+                        throw new UsersNotFoundException("File Users.txt was not found. Make sure it is in the same directory as the .exe file.");
+                    }
+                }
+
+                // Register
+                else if (String.Equals(answer, "register", StringComparison.OrdinalIgnoreCase))
+                {
+                    Dictionary<string, string> usersAndPasswords = GetUsersAndPasswords("Users.txt");
+                    try
+                    {
+                        using (var stream = new StreamWriter("Users.txt", true))
+                        {
+                            while (true)
+                            {
+                                try
+                                {
+                                    string[] credentials = GetCredencials();
+
+                                    SaveToDictionary(usersAndPasswords, credentials);
+                                    stream.WriteLine($"{credentials[0]} {credentials[1]}");
+                                    Console.WriteLine("Hello!");
+                                    break;
+                                }
+                                catch (DuplicateUserCredentialsException ex)
+                                {
+                                    Console.WriteLine($"{ex.Message}\nPlease try again.");
+                                }
+                            }
+                        }
+                    }
+                    catch (FileNotFoundException)
+                    {
+                        throw new UsersNotFoundException("File Users.txt was not found. Make sure it is in the same directory as the .exe file.");
+                    }
+                }
+            }
+            catch (UsersNotFoundException ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+            Console.ReadKey(true);
```

Comment:
I am forced to read through 90 lines to even start getting an idea of what is going on. Please refactor this into at least a few functions.